### PR TITLE
nextcloud-client: clean up

### PR DESCRIPTION
**Pull request by jtojnar**
Continuation of https://github.com/NixOS/nixpkgs/pull/67306

* Switched to non-deprecated dbus config path.
* Removed CMake flags for finding dependencies, the project can find them on its own.
* Properly fixed the RPATH issue from https://github.com/NixOS/nixpkgs/pull/15406 – ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}.
* Dropped changing Icon name in the desktop file as it was fixed upstream.
* Switched to fetchFromGitHub, as the submodules are only for Win, Mac & crash reporting.
* Dropped enableParallelBuilding = true, as it is on by default with CMake.
* Formatted the expression.

cc @Ma27 @worldofpeace

### CODE REVIEW

File: pkgs/applications/networking/nextcloud-client/default.nix [hunk 1]
```diff
@@ -20,29 +20,15 @@ mkDerivation rec {
...
-    "-DCMAKE_BUILD_TYPE=Release"
-    "-DOPENSSL_LIBRARIES=${openssl.out}/lib"
-    "-DOPENSSL_INCLUDE_DIR=${openssl.dev}/include"
-    "-DINOTIFY_LIBRARY=${inotify-tools}/lib/libinotifytools.so"
-    "-DINOTIFY_INCLUDE_DIR=${inotify-tools}/include"
-  ];
+  buildInputs = [ qtbase qtwebkit qtkeychain qttools qtwebengine sqlite openssl pcre inotify-tools libcloudproviders ];
 
   qtWrapperArgs = [
     ''--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libsecret ]}''
```
Comment (worldofpeace) [inline pkgs/applications/networking/nextcloud-client/default.nix]:
```suggestion
    "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libsecret ]}"
```

Funny when you author one case that does this and somehow it's used everywhere :smile:

Comment (jtojnar) [reply pkgs/applications/networking/nextcloud-client/default.nix]: Yeah, it is contagious. I myself was infected by this fad in placeholders :mask:

Comment (worldofpeace) [reply pkgs/applications/networking/nextcloud-client/default.nix]: Right, perhaps when there's actually tooling for the code style it can be a CI check.

**Review by worldofpeace — APPROVED** [review]
Executes on my machine. I'm not a nextcloud user so we should await feedback from @Ma27.

**Review by mmahut — APPROVED** [review]
Thank you for this clean up!

The client looks good, I was able to connect to the nextcloud and sync up files.

**Pull request merged**